redis/db/redisbp/monitored_client.go

```diff
@@ -25,6 +33,9 @@ type PoolStatser interface {
 func NewMonitoredClient(name string, opt *redis.Options) *redis.Client {
 	client := redis.NewClient(opt)
 	client.AddHook(SpanHook{ClientName: name})
+
+	prometheus.MustRegister(NewExporter(client, name))
```

Comment:
same comment about `Must*` here.

looking at the doc, it says:

```
	// If the provided Collector is equal to a Collector already registered
	// (which includes the case of re-registering the same Collector), the
	// returned error is an instance of AlreadyRegisteredError, which
	// contains the previously registered Collector.
	//
	// A Collector whose Describe method does not yield any Desc is treated
	// as unchecked. Registration will always succeed. No check for
	// re-registering (see previous paragraph) is performed. Thus, the
	// caller is responsible for not double-registering the same unchecked
	// Collector, and for providing a Collector that will not cause
	// inconsistent metrics on collection. (This would lead to scrape
	// errors.)
```

so my understanding is that this will always success and never panic. which makes using `MustRegister` fine in this case, but we should add a comment here to explain that.

Reply:
registering an unchecked collector sounds like we're gonna trip over this, can we make it a collector that Describes itself so we _do_ get this checking?